### PR TITLE
FIX: [dca2] must calculate and emit profit at the end of the round

### PR DESCRIPTION
**Pull request by kbearXD**

### CODE REVIEW

File: pkg/strategy/dca2/strategy.go [hunk 1]
```diff
@@ -381,6 +385,26 @@ func (s *Strategy) CalculateAndEmitProfit(ctx context.Context) error {
 		return fmt.Errorf("exchange %s doesn't support ExchangeOrderQueryService", s.ExchangeSession.Exchange.Name())
 	}
 
+	maxTry := 10
+	for try := 1; try < maxTry; try++ {
```
Comment (ycdesu) [inline pkg/strategy/dca2/strategy.go:389]: It retries only 9 times

**Review by ycdesu — APPROVED** [review]

File: pkg/strategy/dca2/state.go [hunk 2]
```diff
@@ -201,7 +201,7 @@ func (s *Strategy) runTakeProfitReady(ctx context.Context, next State) {
 	// reset position
 
 	// calculate profit stats
-	if err := s.CalculateAndEmitProfit(ctx); err != nil {
+	if err := s.mustCalculateAndEmitProfit(ctx); err != nil {
```
Comment (c9s) [inline pkg/strategy/dca2/state.go:204]: when we use "must" it means the function may panic.

File: pkg/strategy/dca2/strategy.go (hunk 1 above)
Comment (c9s) [reply pkg/strategy/dca2/strategy.go:389]: can use backoff here?

**Review by c9s — APPROVED** [review]

File: pkg/strategy/dca2/strategy.go [hunk 3]
```diff
@@ -381,6 +386,22 @@ func (s *Strategy) CalculateAndEmitProfit(ctx context.Context) error {
...
+		}
+
+		if s.ProfitStats.FromOrderID == fromOrderID {
+			return fmt.Errorf("FromOrderID (%d) is not updated, retry it", s.ProfitStats.FromOrderID)
+		}
+
+		return nil
+	}
+
+	return retry.GeneralLiteBackoff(ctx, op)
```
[the remark this answers is not in the document]
Comment (kbearXD) [reply pkg/strategy/dca2/strategy.go:401]: I will print all the rounds and its orders when calculating.

**Pull request merged**